mesonbuild/compilers/compilers.py

```diff
@@ -138,11 +139,15 @@ def is_llvm_ir(fname):
         fname = fname.fname
     return fname.split('.')[-1] == 'll'
 
+@lru_cache(maxsize=None)
+def cached_by_name(fname):
+    suffix = fname.split('.')[-1]
+    return suffix in obj_suffixes
```

Comment:
Could you also try chainging `obj_suffixes` from a tuple to a set? that will make the lookup time log(O) instead of n(O)?

Reply:
That does not really change anything. The tuple is small enough that it is not relevant i guess.